examples/mysql-galera/image/docker-entrypoint.sh

```diff
@@ -0,0 +1,124 @@
+#!/bin/bash
+
+# Copyright 2015 The Kubernetes Authors All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+
+HOSTNAME=`hostname`
+
+if [ "${1:0:1}" = '-' ]; then
+  set -- mysqld "$@"
+fi
+
+if [ "$1" = 'mysqld' ]; then
```

Comment:
I'd prefer to simplify this script if possible, for example purposes. For instance, I think you only use this as an entry point with the mysqld command, so why not get rid of arguments to this script, and always conclude by executing mysqld with the arguments you've set up? Unless there are some best-practices that I'm unaware of here?